### DISCUSSION TITLE
Methods for filtering realtime visits

### DISCUSSION THREAD

**Discussion by joshistoast** [plausible/analytics#2227] [Anything Else]
It's extremely inconvenient that the `realtime` endpoint can only return total realtime visitors for the site and can't be filtered. I'd like to display current visitor counts on specific pages, but the api simply does not allow this yet.

Attempting to work around this, I discovered you can't use the `aggregate` endpoint with 2 `ISO-8601` strings that include a time.

This is invalid:
```
plausible.io/api/v1/stats/aggregate?site_id=[site-id]&filters=event:page==[page-url]&period=custom&date=2022-09-14T16:04:26,2022-09-14T16:09:52
```
> Notice that I'm trying to filter results from the last 5 minutes

This is valid:
```
plausible.io/api/v1/stats/aggregate?site_id=[site-id]&filters=event:page==[page-url]&period=custom&date=2022-09-14,2022-09-14
```

And due to the non-existence of the `filter` parameter on the `realtime` endpoint, I cannot simply do:
```
https://plausible.io/api/v1/stats/realtime/visitors?site_id=[site-id]&filters=event:page==[page-url]
```

It would be a great improvement to expand on the api by allowing times inside `ISO-8601` strings on the `date` parameter, as well as adding the `filter` parameter to the `realtime` endpoint.

**Comment by mattvb91** [comment]
This would be amazing to get in, currently struggling with exactly this. Have you managed to solve it since?

**Comment by polman4** [comment]
I guess this is not resolved somehow yet?
